drivers/sqljs-driver/src/jsMain/kotlin/com/squareup/sqldelight/drivers/sqljs/sqljs.kt

```diff
@@ -0,0 +1,29 @@
+package com.squareup.sqldelight.drivers.sqljs
+
+import org.khronos.webgl.Uint8Array
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlin.js.Promise
+
+operator fun InitStatementJsStatic.invoke(): Statement = createInstance(this)
+operator fun InitDatabaseJsStatic.invoke(): Database = createInstance(this)
+operator fun InitDatabaseJsStatic.invoke(data: Array<Number>): Database = createInstance(this, data)
+operator fun InitDatabaseJsStatic.invoke(data: Uint8Array): Database = createInstance(this, data)
+operator fun InitSqlJsStatic.invoke(): Promise<SqlJsStatic> = asDynamic()()
+operator fun InitSqlJsStatic.invoke(config: Config?): Promise<SqlJsStatic> = asDynamic()(config)
+
+@JsModule("sql.js")
+external val initSqlJs: InitSqlJsStatic
+
+@Suppress("UNUSED_VARIABLE", "UNUSED_PARAMETER")
+fun createInstance(type: dynamic, vararg args: dynamic): dynamic {
+    val argsArray = (listOf(null) + args).toTypedArray()
+    return js("new (Function.prototype.bind.apply(type, argsArray))")
+}
+
+suspend fun initSql(config: Config? = js("{}")): SqlJsStatic = initSqlJs(config).await()
```

Comment:
is there a way we can remove the dependency on coroutines from the driver? It should ideally be all blocking straightforward APIs that we can wrap in coroutines if necessary

Reply:
We would have to expose this as a Promise, which I think is fine. The consumer could then `await()` using the built-in extension for it if they're using coroutines.

Reply:
yea I like that more

Reply:
does `Promise` have a normal blocking `await()`? if so you could probably just use that then in your tests @tokou to get around the issues you're currently seeing

Reply:
No. You would have to do something like
```kotlin
@Test fun myTest() = initDb().then {
  test stuff
}
```
which returns the `Promise` to the test runner which it will then wait on.

Reply:
Done, indeed it's much better now